src/store/useMiningStore.ts

```diff
@@ -35,17 +44,106 @@ const initialState: State = {
     miningInitiated: false,
     isChangingMode: false,
     miningControlsEnabled: true,
+    cpu: {
+        hardware: undefined,
+        mining: {
+            is_mining: false,
+            hash_rate: 0,
+            estimated_earnings: 0,
+            connection: { is_connected: false },
+        },
+    },
+    gpu: {
+        hardware: undefined,
+        mining: {
+            is_mining: false,
+            hash_rate: 0,
+            estimated_earnings: 0,
+        },
+    },
+    base_node: {
+        block_height: 0,
+        block_time: 0,
+        is_synced: false,
+    },
 };
 
-export const useMiningStore = create<MiningStoreState>()((set) => ({
+export const useMiningStore = create<MiningStoreState>()((set, getState) => ({
     ...initialState,
-    setDisplayBlockTime: (displayBlockTime) => set({ displayBlockTime }),
-    setDisplayBlockHeight: (displayBlockHeight) => set({ displayBlockHeight }),
-    setEarnings: (earnings) => set({ earnings }),
+    fetchMiningMetrics: async () => {
+        console.info('Fetching mining metrics...');
+        try {
+            const metrics = await invoke('get_miner_metrics');
+            set(metrics);
+        } catch (e) {
+            console.error(e);
+        }
+    },
+    handleBlockMined: () => {
+        set({
+            postBlockAnimation: true,
+            timerPaused: true,
```

Comment:
```suggestion
            timerPaused: false,
```

was testing and after winning everything would freeze up 😅 button and mode switch disabled, block height wouldn't change, got sooo worried, but it's thankfully just this little value - timer needs to be unpaused hehe

Reply:
Played with this a bit. Refactored calculating `displayBlockTime`. But why do we need to pause timer in a first place? I calculate time: now - last_block_time - it seems like we don't need this anymore

Reply:
we had to add a delay to wait for balance changes because the balance only comes in a bit after the block height changes, without the delay we might only display a win after it moves on to the next block. timer is paused just to allow all the animations to run through before updating the display block height and block time

Reply:
Was it like requirement to reset timer, update balances and display animation in exact the same time? If yes i still think it's better way to handle it: check whether block has changed in store

Reply:
yeah, it was 😅 - https://github.com/tari-project/universe/issues/89 - but that's from a while back, maybe we can change things now? @stringhandler pls advise :D

Reply:
I see, so I can also handle it using current logic, should be better with current flow